src/app/pages/office/cat-b/office.cat-b.page.ts

```diff
@@ -1,58 +1,696 @@
+import { FaultSummaryProvider } from '@providers/fault-summary/fault-summary';
+import { getNewTestStartTime } from '@shared/helpers/test-start-time';
+import { SetStartDate }
+  from '@store/tests/journal-data/common/test-slot-attributes/test-slot-attributes.actions';
+import { Router } from '@angular/router';
+import {
+  OfficeViewDidEnter,
+  CompleteTest,
+  SavingWriteUpForLater,
+  OfficeValidationError,
+  TestStartDateChanged,
+} from '../office.actions';
 import { JOURNAL_PAGE, TestFlowPageNames } from '../../page-names.constants';
+import { behaviourMap } from '../office-behaviour-map';
 
-interface CatBOfficePageState {}
-
-type OfficePageState = CommonOfficePageState & CatBOfficePageState;
+interface OfficePageState {
+  activityCode$: Observable<ActivityCodeModel>;
+  startTime$: Observable<string>;
+  startDate$: Observable<string>;
+  startDateTime$: Observable<string>;
+  testOutcome$: Observable<string>;
+  testOutcomeText$: Observable<string>;
+  isPassed$: Observable<boolean>;
+  isTestOutcomeSet$: Observable<boolean>;
+  candidateName$: Observable<string>;
+  candidateDriverNumber$: Observable<string>;
+  routeNumber$: Observable<number>;
+  displayRouteNumber$: Observable<boolean>;
+  displayIndependentDriving$: Observable<boolean>;
+  displayCandidateDescription$: Observable<boolean>;
+  displayIdentification$: Observable<boolean>;
+  displayShowMeQuestion$: Observable<boolean>;
+  displayTellMeQuestion$: Observable<boolean>;
+  displayWeatherConditions$: Observable<boolean>;
+  displayAdditionalInformation$: Observable<boolean>;
+  displayEco$: Observable<boolean>;
+  displayEta$: Observable<boolean>;
+  displayDrivingFault$: Observable<boolean>;
+  displaySeriousFault$: Observable<boolean>;
+  displayDangerousFault$: Observable<boolean>;
+  identification$: Observable<Identification>;
+  independentDriving$: Observable<IndependentDriving>;
+  candidateDescription$: Observable<string>;
+  additionalInformation$: Observable<string>;
+  showMeQuestion$: Observable<VehicleChecksQuestion>;
+  showMeQuestionOptions$: Observable<VehicleChecksQuestion[]>;
+  tellMeQuestionText$: Observable<string>;
+  etaFaults$: Observable<string>;
+  ecoFaults$: Observable<string>;
+  drivingFaults$: Observable<FaultSummary[]>;
+  drivingFaultCount$: Observable<number>;
+  displayDrivingFaultComments$: Observable<boolean>;
+  weatherConditions$: Observable<WeatherConditions[]>;
+  dangerousFaults$: Observable<FaultSummary[]>;
+  seriousFaults$: Observable<FaultSummary[]>;
+  isRekey$: Observable<boolean>;
+}
 
 @Component({
-  selector: 'app-office-cat-b',
-  templateUrl: './office.cat-b.page.html',
-  styleUrls: ['./office.cat-b.page.scss'],
+  selector: '.office-cat-b-page',
+  templateUrl: 'office.cat-b.page.html',
+  styleUrls: ['../office.page.scss'],
 })
-export class OfficeCatBPage extends OfficeBasePageComponent implements OnInit {
-
+export class OfficeCatBPage extends PracticeableBasePageComponent {
```

Comment:
No use of OfficeBasePageComponent yet, but discussed and going to come in subsequent PR

Reply:
id prefer to move what i can now to the base page so will look into it